components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -3195,6 +3195,11 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         throw new ResponseError(ErrorCodes.SAAS_FEATURE, `Not implemented in this version`);
     }
 
+    async setUsageAttribution(ctx: TraceContext, usageAttributionId: string): Promise<void> {
```

Comment:
How do we roll out this API change? Currently Usage Based Pricing is enabled behind a feature flag.

Reply:
This PR adds a replacement to an unchecked API call, providing more validation. Do you see any value in pushing it behind the feature flag?  e.g. where can it break?

Reply:
The previous way to set the attribution wasn't actually exposed to users in any way, it had to be done through a DB update manually. The difference here is that we're adding an API which is now accessible by users.

If we choose to not gate it behind a feature flag, users will already be able to set their attribution ID to something. That may not necessarily be a bad thing, but it can cause confusion with respect to how Chargebee plans will work with respect to Usage Based Pricing.

My recommendation would be that for now, we gate it behind the `isUsageBasedBillingEnabled` flag, like [here](https://github.com/gitpod-io/gitpod/blob/2946bee541dc1594bc8e80ed1b02f2e9f93eb4d0/components/server/ee/src/workspace/gitpod-server-impl.ts#L1955).

Reply:
> The previous way to set the attribution wasn't actually exposed to users in any way, it had to be done through a DB update manually. The difference here is that we're adding an API which is now accessible by users.

This is not true. `User.additionalData` is open for every user. You may put in there what ever you want.

Reply:
> users will already be able to set their attribution ID to something

Exactly this is what this PR is about. The value should be validated.

Reply:
> This is not true. User.additionalData is open for every user. You may put in there what ever you want

How does a user (either from dashboard, or the API) update this value? I wasn't aware of this being directly exposed to users.

Reply:
> How does a user (either from dashboard, or the API) update this value?
Yes, this used to hold user preferences. I believe this data should only affect this user's resources, which is not true for usage/billing related resources.